tests/func/test_add.py

```diff
@@ -245,66 +246,72 @@ def test_dir(self):
         self.assertEqual(ret, 0)
 
 
-class TestShouldUpdateStateEntryForFileAfterAdd(TestDvc):
-    def test(self):
-        file_md5_counter = spy(dvc.remote.local.file_md5)
-        with patch.object(dvc.remote.local, "file_md5", file_md5_counter):
-            ret = main(["config", "cache.type", "copy"])
-            self.assertEqual(ret, 0)
+class TestDvcWithMocker(TestDvc):
+    @pytest.fixture(autouse=True)
+    def use_mocker(self, mocker):
+        self.mocker = mocker
 
-            ret = main(["add", self.FOO])
-            self.assertEqual(ret, 0)
-            self.assertEqual(file_md5_counter.mock.call_count, 1)
 
-            ret = main(["status"])
-            self.assertEqual(ret, 0)
-            self.assertEqual(file_md5_counter.mock.call_count, 1)
+def test_should_update_state_entry_for_file_after_add(mocker, dvc, tmp_dir):
+    file_md5_counter = mocker.spy(dvc_module.remote.local, "file_md5")
+    tmp_dir.gen("foo", "foo")
 
-            ret = main(["run", "-d", self.FOO, "echo foo"])
-            self.assertEqual(ret, 0)
-            self.assertEqual(file_md5_counter.mock.call_count, 1)
+    ret = main(["config", "cache.type", "copy"])
+    assert ret == 0
 
-            os.rename(self.FOO, self.FOO + ".back")
-            ret = main(["checkout"])
-            self.assertEqual(ret, 0)
-            self.assertEqual(file_md5_counter.mock.call_count, 1)
+    ret = main(["add", "foo"])
+    assert ret == 0
+    assert file_md5_counter.mock.call_count == 1
```

Comment:
can be simplified to `file_md5_counter.call_count`, this applies to each use case of `spy`

Reply:
@pared It is a direct conversion that is explicit, we could leave it as is with `== 1`.

Reply:
Should I change it to not have `.mock` then? It does seem to do the same thing.

Reply:
It does, I don't think its crucial, let's leave it as is.